### PR TITLE
Add an isset test on id_manufacturer while getting product properties

### PR DESCRIPTION
**Pull request by PrestaEdit**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | If the properties doesn't have this index, it will throw a notice.
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no

### CODE REVIEW

**Review by nicosomb — DISMISSED** [review]
LGTM. No need to have an opened issue?

**Review by Hlavtox — CHANGES_REQUESTED** [review]
`$row['manufacturer_name'] = !empty((int) $row['id_manufacturer']) ? Manufacturer::getNameById((int) $row['id_manufacturer']) : null;`

**Comment by mflasquin** [comment]
> `$row['manufacturer_name'] = !empty((int) $row['id_manufacturer']) ? Manufacturer::getNameById((int) $row['id_manufacturer']) : null;`

In the commit of @PrestaEdit there is an additional test to verify if $row['id_manufacturer'] > 0. I don't know if this check is really mandatory. What do you think about that @PrestaEdit ?

**Comment by PrestaEdit** [comment]
As empty() will check if the var exists or it's falsey (0, '0', it's not needed).

**Comment by Hlavtox** [comment]
@mflasquin That check is not needed. `!empty` returns true only if the variable exists, is not NULL, not zero, not string zero, not empty array, empty string etc. 👍

@PrestaEdit you are faster :D

**Comment by mflasquin** [comment]
> As empty() will check if the var exists or it's falsey (0, '0', it's not needed).

Ok so I think you can follow @Hlavtox 's suggestion :)

**Review by mflasquin — COMMENTED** [review]
I'm ok with the updated code but I think you maybe have to indicate how to test to help the QA to validate your PR :wink:

**Review by mflasquin — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
IMHO it doesn't need QA.

File: classes/Product.php [hunk 1]
```diff
@@ -5512,7 +5512,7 @@ public static function getProductProperties($id_lang, $row, Context $context = n
         $row['category'] = Category::getLinkRewrite((int) $row['id_category_default'], (int) $id_lang);
         $row['category_name'] = Db::getInstance()->getValue('SELECT name FROM ' . _DB_PREFIX_ . 'category_lang WHERE id_shop = ' . (int) $context->shop->id . ' AND id_lang = ' . (int) $id_lang . ' AND id_category = ' . (int) $row['id_category_default']);
         $row['link'] = $context->link->getProductLink((int) $row['id_product'], $row['link_rewrite'], $row['category'], $row['ean13']);
-        $row['manufacturer_name'] = (int) $row['id_manufacturer'] > 0 ? Manufacturer::getNameById((int) $row['id_manufacturer']) : null;
+        $row['manufacturer_name'] = !empty((int) $row['id_manufacturer']) ? Manufacturer::getNameById((int) $row['id_manufacturer']) : null;
```
Comment (lartist) [inline classes/Product.php:5515]:
```suggestion
        $row['manufacturer_name'] = !empty($row['id_manufacturer']) ? Manufacturer::getNameById((int) $row['id_manufacturer']) : null;
```

**Review by FabienPapet — APPROVED** [review]

**Comment by kpodemski** [comment]
@PrestaShop/qa-team IMHO it doesn't need QA, it's a safe change to avoid potential PHP notice, please confirm :-)

**Comment by hibatallahAouadni** [comment]
Hello @kpodemski 

I just ran the auto test, if it's all green we can merge it :wink: 
Here's the link: https://github.com/hibatallahAouadni/testing_pr/actions/runs/3435876124

Thanks!

**Comment by kpodemski** [comment]
Thanks @hibatallahAouadni :)

Thanks @PrestaEdit

**Pull request merged**